### ISSUE TITLE
SelfPublishModel causes all ForeignKey queries of model to be exectued

### ISSUE THREAD

**Issue by Gromph** [jonashagstedt/swampdragon#52]
As we added the SelfPublishModel to more of our existing models we started noticing a massive increase in the number of queries to our database server. It appears that when SelfPublishModel saves the state of a ForeignKey field it triggers the query to retrieve the related model from the db. It seems like it should just need to save the primary key.

For now we've had to back off on how many models we add SelfPublishModel to until we can figure out how to get these excess queries under control.

Thanks!

**Comment by Gromph** [comment]
I made a temporary solution by sub classing SelfPublishModel and replacing _get_relevant_fields with the following function. Bascially it replaces ForeignKeys field names, with the name of the key, ie swap out field for field_id

One view I was testing went from 25000 queries to 25

```python
    def _get_relevant_fields(self):
        """
        Get all fields that will affect the state.
        This is used to save the state of the model before it's updated,
        to be able to get changes used when publishing an update (so not all fields are published)
        """
        # update_fields = list(self._serializer.opts.update_fields)
        # publish_fields = list(self._serializer.opts.publish_fields)
        # relevant_fields = set(update_fields + publish_fields)
        relevant_fields = self._serializer.base_fields

        if 'id' in relevant_fields:
            relevant_fields.remove('id')

        relevant_fields_copy = []
		
        # check for any foreign keys and include the key to the object instead of the object itself
        # This is to avoid triggering queries to retrieve the foreign object
        for field in self._meta.fields:
            if field.name in relevant_fields:
                if type(field) in (models.ForeignKey, ):
                    # typically field.attname is field.name + "_id" ie field_id
                    relevant_fields_copy.append(field.attname)
                else:
                    relevant_fields_copy.append(field.name)

        return relevant_fields_copy
``` 

I'm not sure if this is the best permanent solution though.

**Comment by jonashagstedt** [comment]
Looking at this now to see if I can get this into the 0.4.2 release which I'm aiming to get out this week

**Comment by jonashagstedt** [comment]
This is not implemented in the 0.4.2 branch if you want to give it a go.
